File: src/main/java/ru/nucodelabs/gem/view/AbstractController.java
package ru.nucodelabs.gem.view;

import javafx.fxml.Initializable;
import javafx.stage.Stage;

/**
 * Контроллер, который не только инициализируется FXML-загрузчиком, но и знает как добыть свою сцену
 */
public abstract class AbstractController implements Initializable {
    protected abstract Stage getStage();
}

File: src/main/java/ru/nucodelabs/gem/view/AbstractController.kt
package ru.nucodelabs.gem.view

import javafx.fxml.Initializable
import javafx.stage.Stage

/**
 * Контроллер, который не только инициализируется FXML-загрузчиком, но и знает как добыть свою сцену
 */
abstract class AbstractController : Initializable {
    protected abstract val stage: Stage?
}